src/main/scala/is/hail/expr/ir/Compile.scala

```diff
@@ -11,18 +12,18 @@ object Compile {
   def apply[F >: Null : TypeInfo, R: TypeInfo : ClassTag](args: Seq[(String, Type, ClassTag[_])], body: IR): (Type, () => F) = {
     assert(args.forall{ case (_, t, ct) => TypeToIRIntermediateClassTag(t) == ct })
 
-    val argTypeInfo: Array[GenericTypeInfo[_]] =
+    val argTypeInfo: Array[MaybeGenericTypeInfo[_]] =
       GenericTypeInfo[Region]() +:
         args.flatMap { case (_, t, _) =>
-          List[GenericTypeInfo[_]](GenericTypeInfo()(typeToTypeInfo(t)), GenericTypeInfo[Boolean]()).iterator
+          Seq[GenericTypeInfo[_]](GenericTypeInfo()(typeToTypeInfo(t)), GenericTypeInfo[Boolean]())
         }.toArray
 
-    val fb = new FunctionBuilder[F](argTypeInfo.asInstanceOf[Array[MaybeGenericTypeInfo[_]]], GenericTypeInfo[R]())
+    val fb = new FunctionBuilder[F](argTypeInfo, GenericTypeInfo[R]())
 
     var ir = body
     val env = args
       .zipWithIndex
-      .foldLeft(new Env[IR]()) { case (e, ((n, t, _), i)) => e.bind(n, In(i, t)) }
+      .foldLeft(Env.empty[IR]) { case (e, ((n, t, _), i)) => e.bind(n, In(i, t)) }
```

Comment:
shouldn't we shift the In's index by one so that the first index is still the aggregable?